### PR TITLE
fix: Check if on_demand_feature_views is an empty list rather than None for snowflake provider

### LINKED ISSUE #3045: Wrong condition in to_snowflake method of snowflake provider
## Expected Behavior 
[to_snowflake](https://github.com/feast-dev/feast/blob/master/sdk/python/feast/infra/offline_stores/snowflake.py#L430) should ignore on_demand_feature_views if on_demand_feature_views is an empty list

## Current Behavior

to_snowflake runs on_demand_feature_views code because on_demand_feature_views is set to [] [here](https://github.com/feast-dev/feast/blob/master/sdk/python/feast/infra/offline_stores/snowflake.py#L387)

### Specifications

- Version: 0.23.0
- Platform: feast SDK
- Subsystem:

## Possible Solution

`if self.on_demand_feature_views is not None: ` => `if self.on_demand_feature_views is not []: `

### PR DESCRIPTION
**Pull request by floOnfido**
**What this PR does / why we need it**:

**Which issue(s) this PR fixes**:

Fixes #3045

### CODE REVIEW

**Comment by adchia** [comment]
Thanks for the PR!

Can you sign your commits please?

File: sdk/python/feast/infra/offline_stores/snowflake.py [hunk 1]
```diff
@@ -429,7 +429,7 @@ def _to_arrow_internal(self) -> pa.Table:
 
     def to_snowflake(self, table_name: str, temporary=False) -> None:
         """Save dataset as a new Snowflake table"""
-        if self.on_demand_feature_views is not None:
+        if self.on_demand_feature_views is not []:
```
Comment (achals) [inline sdk/python/feast/infra/offline_stores/snowflake.py:432]:
```suggestion
        if self.on_demand_feature_views:
```

**Comment by adchia** [comment]
you'll also need to sign your commit btw!

**Comment by floOnfido** [comment]
Should be good now @adchia

**Review by adchia — APPROVED** [review]

**Pull request merged**